File: rec-core/src/main/java/net/kimleo/rec/rule/RuleLoader.java
package net.kimleo.rec.rule;

import net.kimleo.rec.repository.selector.Selector;

import java.util.List;
import java.util.stream.Collectors;

public class RuleLoader {

    public final RuleRepository repository = new RuleRepository();

    public List<RuleRunner> load(List<String> lines) {
        return lines.stream().map(line -> {
            String[] pair = line.split(":");
            String rule = pair[0];
            String selector = pair[1];

            return new RuleRunner(repository.rule(rule), Selector.of(selector));
        }).collect(Collectors.toList());
    }
}

File: rec-core/src/main/kotlin/net/kimleo/rec/rule/RuleLoader.kt
package net.kimleo.rec.rule

import net.kimleo.rec.repository.selector.Selector

class RuleLoader(val ruleRepository: RuleRepository = RuleRepository()) {
    fun load(lines: List<String>): List<RuleRunner> {
        return lines.map { line ->
            val split = line.split(":")
            val rule = split[0].trim()
            val selector = split[1].trim()

            RuleRunner(ruleRepository.rule(rule), Selector.of(selector))
        }
    }
}

